### PR TITLE
#293 support limit query param in /txns.json

### LINKED ISSUE #293: /txns API limit param
The list of transactions (`/txns` API) may be too long and fetched from the server in many seconds. (715 transactions are loaded ~17 sec for me). Quite often I need only few last transactions to see, so I'd suggest adding query parameter `limit` to select only head of transactions: `/txns?limit=10`.

### PR DESCRIPTION
**Pull request by edmoffo**
Closes #293.

The `/txns.json` route serialised the full wallet transaction history on every call, which the linked issue measured at roughly 17 seconds for a 715-transaction wallet when only the most recent few rows were needed.

The change accepts an optional `limit` query parameter, parses it as an integer after the optional `sort=desc` reverse, and slices the list down to the first `limit` entries before serialising. A non-positive value short-circuits with `WTS::UserError` E207 so a typo never silently returns the entire history.

Locally `ruby -c wts.rb` reports `Syntax OK` and `bundle exec rubocop wts.rb` reports no offences against the project config. The full rake target was not executed in this environment because it requires a live PostgreSQL pool via `DATABASE_URL`; the existing smoke test in `test/test_wts.rb` already hits `/txns.json` without a `limit` parameter and the new branch leaves that no-parameter path unchanged.

### CODE REVIEW

File: wts.rb [hunk 1]
```diff
@@ -258,6 +258,11 @@
   confirmed_user.wallet do |wallet|
     list = wallet.txns
     list.reverse! if params[:sort] && params[:sort] == 'desc'
+    if params[:limit]
+      limit = params[:limit].to_i
+      raise WTS::UserError, "E207: The 'limit' parameter must be a positive integer" if limit <= 0
```
Comment (kreinba) [inline wts.rb:263]: Error code E207 is already taken by front/front_upwork.rb:26, which raises WTS::UserError on a different condition (`You have to work in Zerocracy in order to cash out to UpWork`). Reusing the same code on a new failure path breaks the contract that callers and operators rely on to identify which user-facing error fired. Pick the next free code in the sequence; E227 appears to be the next unused slot after E226 in objects/ops.rb.

File: wts.rb [hunk 2]
```diff
@@ -258,6 +258,11 @@
   confirmed_user.wallet do |wallet|
     list = wallet.txns
     list.reverse! if params[:sort] && params[:sort] == 'desc'
+    if params[:limit]
```
Comment (kreinba) [inline wts.rb:261]: The new limit branch is not exercised by any test. The smoke test in test/test_wts.rb only hits /txns.json without a parameter, so a regression in the parse, the bound check, or the first(limit) slice would land silently. Add a case that calls /txns.json?limit=1 and asserts the response is a JSON array of length at most one, and a case that asserts a 4xx response for limit=0 or a negative value so the new WTS::UserError is covered.

**Review by kreinba — CHANGES_REQUESTED** [review]
Request changes. The reused E207 error code clashes with the cash-out path in front/front_upwork.rb:26 and must be renumbered before merge; the new limit branch also ships without a test for either the happy path or the rejected-zero path. Two inline comments below name the two defects. The two failing CI checks (typos on objects/referrals.rb and yamllint on .github/workflows/actionlint.yml) point at files outside this diff and so are not caused by this change.
